### ISSUE TITLE
Blazor breaking change in 7.0.302 SDK?

### ISSUE THREAD

**Issue by ScarletKuro** [dotnet/aspnetcore#48267]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Hello,

After updating the SDK to version 7.0.302, we encountered numerous errors in our MudBlazor project. One particular example is the ambiguous call issue shown in the screenshot below. It appears that the Razor parser has become more strict, and the unnecessary "@" symbol may be causing the problem. However, I'm unsure how to resolve all the errors, such as "The attribute names could not be inferred from bind attribute 'bind-IsVisible'" or "Multiple components use the tag 'MudSwitch'." Could you provide some insight into their meaning?

While I don't have a reproduction repository available, you can review the logs at the following link for further details:
[GitHub Logs](https://github.com/MudBlazor/MudBlazor/actions/runs/4995425079/jobs/8948000936?pr=6860)

If you clone the MudBlazor repository and compile it with the **7.0.302** SDK version, you'll have a better understanding of the overall situation.

It would be greatly appreciated if you could provide any information or documentation regarding the changes and how to address them. Are there any workarounds available? I'm also curious why a breaking change was introduced in the .NET 7 SDK rather than do them in .NET 8.

Thank you for your assistance.

### .NET Version

7.0.302

### Anything else?

![ambigious1](https://github.com/dotnet/aspnetcore/assets/19953225/1d47dde5-72c7-4a45-87de-eba7ead25652)

**Comment by sofiageo** [comment]
I have the same issue with .NET 8 preview 4 and my project.

**Comment by SpikeViper** [comment]
Having issues with 7.0.302 as well. 
Getting 
`error CS0400: The type or namespace name 'InputComponent' could not be found in the global namespace (are you missing an assembly reference?) `
But the imports are definitely there... and it works perfectly in previous versions. Not sure if related to this issue, but certainly related to the same version change.

**Comment by c4l3b** [comment]
I've experienced a handful of similar issues after installing 7.0.302 (including this one specifically)

For the `could not be found in the global namespace` issues, if I refactored the not-found, single file component to a component with a separate code-behind class, it fixed the issue. Bizarre.

**Comment by SpikeViper** [comment]
> I've experienced a handful of similar issues after installing 7.0.302 (including this one specifically)
> 
> For the `could not be found in the global namespace` issues, if I refactored the not-found, single file component to a component with a separate code-behind class, it fixed the issue. Bizarre.

Thanks for that info. Are you by any chance using components with inheritance? I think that may be related.

**Comment by c4l3b** [comment]
Yes! The issue that I can't resolve right now involves an inherited component with a render fragment. The consumer cannot resolve `context` in the render fragment

**Comment by ScarletKuro** [comment]
> The consumer cannot resolve context in the render fragment

I think we have this issue as well.

**Comment by c4l3b** [comment]
I was going down the rabbit hole of trying to fix it, but I think I'm going to lock the version to 7.0.202 in my global.json file and wait and see if Microsoft issues a fix. Sounds like some unintended breaking changes...

**Comment by SpikeViper** [comment]
Looks like the issue is related to passing 'this' in a blazor component to a subcomponent in my case... that's where I'm having this pop up.
